docs/adr/0007-seperate-binding-generators.md

```diff
@@ -0,0 +1,142 @@
+
+    This means that `uniffi_bindgen` would still handle generic tasks related to binding generation. This includes:
+        1. Parsing the UDL, and passing the `ComponentInterface` to the specific binding generator
+        1. Calling a generic `BindingGenerator::write_bindings` function that is implemented by the specific binding generator
+        1. `uniffi_bindgen` would also expose its `CodeOracle`, `CodeType` and `CodeDeclaration` types to help developers create a standard way to interact with code generation. It shouldn't however, restrict developers to use those concepts if they don't fit a specific binding.
+    But developers would need a separate crate that implements the traits `uniffi_bindgen` exposes.
+
+* **[Option 3] Remove `uniffi_bindgen`, replace it with `uniffi_scaffolding` and have each binding generator implementation fully manage the generation of bindings**
+    This means external crates would have to depend on `uniffi` directly and implement command-line interfaces. This gives full autonomy to the binding generator developer. Developers won't have access to any traits from `uniffi_bindgen`
+
+## Pros and Cons of the Options
+
+### **[Option 1] Do nothing**
+
+* Good, because it keeps version compatibility easy to guarantee.
+* Good, it makes it easier to make changes to multiple bindings at a time (in the case of a breaking change in `uniffi`, etc)
+* Bad, because maintainability can grow to be difficult - especially if more bindings are added which the core `uniffi` team is not familiar with
+* Bad, Because testability can also grow to be difficult - as more bindings are added (some type of `gecko-js` is expected in the future) the requirement to test all the bindings together in one repository is difficult to maintain.
+* Bad, Because releases of all the binding generators are tied to the release of `uniffi_bindgen`
+
+### **[Option 2] Expose a trait in `uniffi_bindgen` that can be implemented by other crates, to separate the implementation of the bindings**
+
+* Good, because maintainability will be clear, and members of the community can opt to maintain their own binding generators
+* Good, because testability can improve, where a developer can test a binding generator without building or testing the others
+* Good, because our CI would only need to test the core bindings we maintain, and others can be tested by their own maintainers (for example, a `gecko-js` binding generator should be tested in `mozilla-central` and not here)
+* Good, because a release in one binding wouldn't have an impact on any other ones
+* Bad, because it complicates version compatibility, and we will need more assertions to guarantee it
+* Bad, because testability for might increase in complexity. Bindings not in this repository will not have access to the extensive fixtures and examples we have. We might need to expose those if take this option.
```

Comment:
```suggestion
* Bad, because testability might increase in complexity. Bindings not in this repository will not have access to the extensive fixtures and examples we have. We might need to expose those if take this option.
```